### PR TITLE
[NEAT-297, NEAT-298]🐳 Support Edges with Properties

### PR DESCRIPTION
**Pull request by doctrino**
A few observations:
1. We likely have a bug with inwards edges. There is an assumption that if the outwards edge is single edge, then the inwards have to be the same. This does not hold, as multiple instances can have a single edge to a parent, that has inwards edges. This is not very serious as single and multi edge is not enforced by the API.
2. The way we deal with default space/prefix/version is painful. Lost a lot of time to find it deep in the serialization. Need to consider whether adding and removing default space is a set with Transformers. @nikokaoja Wdyt?
3. The DMSExporter should have a small refactoring it was cumbersome to work with. The DMSImporter on the other hand was a dream :) 

image

image

### CODE REVIEW

**Review by nikokaoja — APPROVED** [review]
See couple of my comments

File: cognite/neat/rules/importers/_dms2rules.py [hunk 1]
```diff
@@ -335,7 +337,7 @@ def _create_dms_property(
             property_=prop_id,
             description=prop.description,
             name=prop.name,
-            connection=self._get_relation_type(prop),
+            connection=self._get_connection_type(prop),
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/rules/importers/_dms2rules.py]: yes

File: cognite/neat/rules/importers/_dms2rules.py [hunk 2]
```diff
@@ -455,3 +468,41 @@ def _get_constraint(self, prop: ViewPropertyApply, prop_id: str) -> list[str] |
...
         return unique_constraints or None
+
+    def _find_reverse_edge(
+        self, prop_id: str, prop: SingleEdgeConnectionApply | MultiEdgeConnectionApply, view_id: dm.ViewId
+    ) -> str | None:
+        if prop.source not in self._all_views_by_id:
+            return None
+        view = self._all_views_by_id[prop.source]
+        candidates = []
+        for prop_name, reverse_prop in (view.properties or {}).items():
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/rules/importers/_dms2rules.py:479]: yes

File: cognite/neat/rules/importers/_dms2rules.py [hunk 3]
```diff
@@ -455,3 +468,41 @@ def _get_constraint(self, prop: ViewPropertyApply, prop_id: str) -> list[str] |
                     )
                 )
         return unique_constraints or None
+
+    def _find_reverse_edge(
+        self, prop_id: str, prop: SingleEdgeConnectionApply | MultiEdgeConnectionApply, view_id: dm.ViewId
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/rules/importers/_dms2rules.py:473]: no, you cannot. It is a dict and the prop id is the key. This is who it comes from the API

File: cognite/neat/rules/importers/_dms2rules.py [hunk 4]
```diff
@@ -371,18 +373,21 @@ def _get_relation_type(self, prop: ViewPropertyApply) -> Literal["edge", "revers
 
     def _get_value_type(
         self, prop: ViewPropertyApply, view_entity: ViewEntity, prop_id
-    ) -> DataType | ViewEntity | ViewPropertyEntity | DMSUnknownEntity | None:
-        if isinstance(prop, SingleEdgeConnectionApply | MultiEdgeConnectionApply) and prop.direction == "outwards":
-            return ViewEntity.from_id(prop.source)
+    ) -> DataType | ViewEntity | EdgeViewEntity | ViewPropertyEntity | DMSUnknownEntity | None:
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/rules/importers/_dms2rules.py]: No, only edge

File: cognite/neat/rules/models/dms/_rules.py [hunk 5]
```diff
@@ -145,8 +146,8 @@ class DMSProperty(SheetEntity):
     view_property: str = Field(alias="View Property")
     name: str | None = Field(alias="Name", default=None)
     description: str | None = Field(alias="Description", default=None)
-    connection: Literal["direct", "edge", "reverse"] | None = Field(None, alias="Connection")
-    value_type: DataType | ViewPropertyEntity | ViewEntity | DMSUnknownEntity = Field(alias="Value Type")
+    connection: Literal["direct"] | ReverseEntity | EdgeEntity | None = Field(None, alias="Connection")
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/rules/models/dms/_rules.py]: You cannot have any extra information about a direct relations, so not sure it makes sense.

**Pull request merged**